src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/study/StudyAdherenceReportGeneratorTest.java

```diff
@@ -1359,10 +1369,298 @@ public void dateRangeErrorDefaultsToStreamDateRange() {
+        assertStartDate(weeklyReport, 5, LocalDate.parse("2022-04-13"));
+        assertStartDate(weeklyReport, 6, LocalDate.parse("2022-04-14"));
+    }
+    
+    // BRIDGE-3288 + BRIDGE-3287
+    @Test
+    public void negativeDaysDoNotBreakScheduleSimplified() throws JsonProcessingException {
+        // This schedule won't pass validation, but I'm concerned there are “legal” ways
+        // to break the schedule when there are overlaps and negative weeks.
+        AssessmentReference ref1 = new AssessmentReference();
+        ref1.setGuid("survey");
+        ref1.setAppId(TEST_APP_ID);
+        ref1.setIdentifier("survey");
+        
+        TimeWindow win1 = new TimeWindow();
+        win1.setGuid(TestConstants.SESSION_WINDOW_GUID_1);
+        win1.setStartTime(LocalTime.parse("08:00"));
+        
+        Session session1 = new Session();
+        session1.setGuid(SESSION_GUID_1);
+        session1.setName("session1");
+        session1.setAssessments(ImmutableList.of(ref1));
+        session1.setInterval(Period.parse("P1W"));
+        session1.setTimeWindows(ImmutableList.of(win1));
+        session1.setStartEventIds(ImmutableList.of("custom:event1"));
+        
+        TimeWindow win2 = new TimeWindow();
+        win2.setGuid(SESSION_WINDOW_GUID_2);
+        win2.setStartTime(LocalTime.parse("08:30"));
+        
+        Session session2 = new Session();
+        session2.setGuid(SESSION_GUID_2);
+        session2.setName("session2");
+        session2.setAssessments(ImmutableList.of(ref1));
+        session2.setInterval(Period.parse("P1W"));
+        session2.setTimeWindows(ImmutableList.of(win2));
+        session2.setStartEventIds(ImmutableList.of("custom:event2"));
+
+        Schedule2 schedule = new Schedule2();
+        schedule.setGuid(SCHEDULE_GUID);
+        schedule.setDuration(Period.parse("P2W"));
+        schedule.setSessions(ImmutableList.of(session1, session2));
+        
+        Timeline timeline = Scheduler.INSTANCE.calculateTimeline(schedule);
+        
+        StudyActivityEvent event1 = new StudyActivityEvent.Builder()
+                .withEventId("custom:event1")
+                .withTimestamp(DateTime.parse("2022-04-05T12:00:00.000Z")).build();
+        StudyActivityEvent event2 = new StudyActivityEvent.Builder()
+                .withEventId("custom:event2")
+                .withTimestamp(DateTime.parse("2022-04-20T12:00:00.000Z")).build();
+        
+        AdherenceState state = new AdherenceState.Builder()
+                .withMetadata(timeline.getMetadata())
+                .withEvents(ImmutableList.of(event1, event2))
+                .withNow(DateTime.parse("2022-04-24T12:00:00.000Z"))
+                .withStudyStartEventId("custom:event2")
+                .build();
+            
+        StudyAdherenceReport report = StudyAdherenceReportGenerator.INSTANCE.generate(state);
+        
+        StudyReportWeek week = report.getWeeks().get(0);
+        assertEquals(week.getWeekInStudy(), -1);
```

Comment:
This is the off-by-one error, shouldn't this be Week -2?

Reply:
No. Notice that the study start event on 4/20, is correctly located in Week 1. This is correct (it doesn't have to be on day 0 of the week).